src/app/rpmostree-builtin-initramfs.c

```diff
@@ -0,0 +1,153 @@
+
+  if (!rpmostree_load_os_proxy (sysroot_proxy, NULL,
+                                cancellable, &os_proxy, error))
+    goto out;
+
+  if (!(opt_enable || opt_disable))
+    {
+      GVariantIter iter;
+      g_autoptr(GVariant) deployments = rpmostree_sysroot_dup_deployments (sysroot_proxy);
+
+      g_variant_iter_init (&iter, deployments);
+
+      while (TRUE)
+        {
+          gboolean cur_regenerate;
+          g_autoptr(GVariant) child = g_variant_iter_next_value (&iter);
+          g_autoptr(GVariantDict) dict = NULL;
+          g_autofree char **initramfs_args = NULL;
+          gboolean is_booted;
+
+          if (child == NULL)
+            break;
+
+          dict = g_variant_dict_new (child);
+
+          if (!g_variant_dict_lookup (dict, "booted", "b", &is_booted))
+            continue;
+          if (!is_booted)
+            continue;
+
+          if (!g_variant_dict_lookup (dict, "regenerate-initramfs", "b", &cur_regenerate))
+            cur_regenerate = FALSE;
+          if (cur_regenerate)
+            {
+              g_variant_dict_lookup (dict, "initramfs-args", "^a&s", &initramfs_args);
+            }
+
+          g_print ("Initramfs regeneration: %s\n", cur_regenerate ? "enabled" : "disabled");
+          if (initramfs_args)
+            {
+              g_print ("Initramfs args: ");
+              for (char **iter = initramfs_args; iter && *iter; iter++)
+                g_print ("%s ", *iter);
+              g_print ("\n");
+            }
+        }
+    }
+  else if (opt_enable && opt_disable)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "Cannot simultaenously specify --enable and --disable");
+      goto out;
+    }
+  else
+    {
+      char *empty_strv[] = {NULL};
+      if (opt_disable && opt_add_arg)
+        {
+          g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                       "Cannot simultaenously specify --disable and --arg");
+          goto out;
+        }
+      if (!opt_add_arg)
+        opt_add_arg = empty_strv;
+      if (!rpmostree_os_call_set_initramfs_state_sync (os_proxy,
+                                                       opt_enable,
+                                                       (const char *const*)opt_add_arg,
+                                                       g_variant_builder_end (opts_builder),
+                                                       &transaction_address,
+                                                       cancellable,
+                                                       error))
+        goto out;
+
+      if (!rpmostree_transaction_get_response_sync (sysroot_proxy,
+                                                    transaction_address,
+                                                    cancellable,
+                                                    error))
+        goto out;
+
+      g_print ("Initramfs regeneration is now: %s\n", opt_enable ? "enabled" : "disabled");
```

Comment:
The tricky thing about this is that right after this is printed out, the user will want to do `rpm-ostree initramfs` to confirm the new state and will be disappointed to see that it's not what they expected.

I would consider changing this to e.g.: "[Enabled|Disabled] initramfs regeneration in new deployment.". That, together with the customary "Run \"systemctl reboot\" to start a reboot\n" will make it more obvious what's going on.